File: frontend/src/services/adminDashboardService.js
import api from './api';

/**
 * Admin Dashboard API Service
 * Layer: Service (API calls)
 * Handles all HTTP requests for admin dashboard metrics and alerts
 */

/**
 * Fetch admin dashboard metrics (KPIs and alerts)
 * @returns Promise<AdminDashboardResponse> containing dashboard data
 * @throws Error if unauthorized (403) or unauthenticated (401)
 */
export const getDashboardMetrics = async () => {
  try {
    const response = await api.get('/api/admin/dashboard');
    return response.data;
  } catch (error) {
    console.error('Error fetching dashboard metrics:', error);
    throw error;
  }
};

export default {
  getDashboardMetrics,
};

File: backend/src/main/java/com/urbanfresh/service/AdminDashboardService.java
package com.urbanfresh.service;

import com.urbanfresh.dto.AdminDashboardResponse;

/**
 * Admin Dashboard Service Interface
 * Layer: Service (Business Logic)
 * Provides KPI metrics and alert counts for admin dashboard
 */
public interface AdminDashboardService {
    
    /**
     * Retrieves admin dashboard metrics and alerts
     * @return AdminDashboardResponse containing KPIs and alert counts
     */
    AdminDashboardResponse getDashboardMetrics();
}
